### PR TITLE
Migrate to Vue 3 Migration Build

### LINKED ISSUE #5713: Notebook should use the Menu API for context menus
The Notebook object has a number of context menus which look and feel like menus generated from the Menu API, but are really generated by a [PopupMenu component exclusive to the Notebook](https://github.com/nasa/openmct/blob/0f12aa1eaebe25bee0daed039ac45e083ce637f4/src/plugins/notebook/components/PopupMenu.vue), whose existence predates the Menu API.

The Notebook should be updated to use the Menu API to help ensure consistent context menu behavior across the application.

### LINKED ISSUE #6721: Upgrade to Vue 3 Migration Build
#### Summary

- Upgrade Open MCT to the [Vue 3 Migration Build.](https://v3-migration.vuejs.org/migration-build.html).
- Provide documentation on migration process and any caveats / limitations.
- Identify steps to migrate away from the combatibility version to Vue 3 proper.

### PR DESCRIPTION
**Pull request by ozyx**
Closes #6721 
Closes #5713  

### Describe your changes:

This PR contains the changes necessary to upgrade Open MCT from Vue 2 to the [Vue 3 Migration Build](https://v3-migration.vuejs.org/migration-build.html). It resolves all [incompatibilities](https://v3-migration.vuejs.org/migration-build.html#incompatible) between Vue 2 and this migration build. Some of the notable changes include:

- Replacing all instances of the `new Vue()` component creation pattern
  - In Vue 3, components cannot be created on the fly and mounted off-DOM. The suggested fix from Vue is to use `createApp`, but in the context of Open MCT this means dozens of Vue apps being created and destroyed at any given moment. Instead, we have used a [community hack ](https://github.com/nasa/openmct/blob/vue3-compat-migration/src/utils/mount.js)for creating individual components.
- `beforeDestroy()` -> `beforeUnmount()`
- `destroyed()` -> `unmounted()`
- The addition of `deep: true` option on Array listeners is now required to detect Array changes
- Open MCT is now mounted on a child `div` instead of directly on `document.body`

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [ ] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [ ] Changes address original issue?
* [ ] Tests included and/or updated with changes?
* [ ] Command line build passes?
* [ ] Has this been smoke tested?
* [ ] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [x] Code style and in-line documentation are appropriate?
* [x] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: example/simpleVuePlugin/plugin.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ function SimpleVuePlugin() {
             container.appendChild(vm.$mount().$el);
           },
           destroy: function (container) {
-            vm.$destroy();
+            //vm.$destroy();
```
Comment (scottbell) [inline example/simpleVuePlugin/plugin.js:27]: Heh, looks like it basically was just [renamed](https://v3-migration.vuejs.org/breaking-changes/)!

> The destroyed lifecycle option has been renamed to unmounted

File: src/api/composition/CompositionCollection.js [hunk 2]
```diff
@@ -56,10 +56,10 @@
  */
 export default class CompositionCollection {
   domainObject;
-  #provider;
-  #publicAPI;
-  #listeners;
-  #mutables;
+  provider;
```
Comment (scottbell) [inline src/api/composition/CompositionCollection.js:59]: Why do we need to make these public?

File: src/api/composition/DefaultCompositionProvider.js [hunk 3]
```diff
@@ -167,7 +168,7 @@ export default class DefaultCompositionProvider extends CompositionProvider {
    */
   add(parent, childId) {
     if (!this.includes(parent, childId)) {
-      const composition = structuredClone(parent.composition);
+      const composition = structuredClone(toRaw(parent.composition));
```
Comment (scottbell) [inline src/api/composition/DefaultCompositionProvider.js:171]: [Cool](https://vuejs.org/api/reactivity-advanced.html#toraw)

File: src/plugins/condition/components/inspector/StylesView.vue [hunk 4]
```diff
@@ -878,8 +878,8 @@ export default {
         const fontSize = hasConsolidatedFontSize ? styles[0].fontSize : NON_SPECIFIC;
         const font = hasConsolidatedFont ? styles[0].font : NON_SPECIFIC;
 
-        this.$set(this.consolidatedFontStyle, 'fontSize', fontSize);
-        this.$set(this.consolidatedFontStyle, 'font', font);
+        this.consolidatedFontStyle.fontSize = fontSize;
```
Comment (scottbell) [inline src/plugins/condition/components/inspector/StylesView.vue:881]: this is wonderful

File: src/plugins/folderView/components/ListItem.vue [hunk 5]
```diff
@@ -21,6 +21,7 @@
 -->
 <template>
   <tr
+    ref="root"
```
Comment (scottbell) [inline src/plugins/folderView/components/ListItem.vue:24]: what's this for?

File: src/plugins/notebook/components/SectionCollection.vue [hunk 6]
```diff
@@ -78,10 +78,13 @@ export default {
...
-      if (!this.containsSection(this.selectedSectionId)) {
-        this.selectSection(this.sections[0].id);
-      }
+    sections: {
+      handler(val, oldVal) {
+        if (!this.containsSection(this.selectedSectionId)) {
+          this.selectSection(this.sections[0].id);
+        }
+      },
+      deep: true
```
Comment (scottbell) [inline src/plugins/notebook/components/SectionCollection.vue:87]: Ah, [another breaking change](https://v3-migration.vuejs.org/breaking-changes/).

> [When watching an array, the callback will only trigger when the array is replaced. If you need to trigger on mutation, the deep option must be specified.](https://v3-migration.vuejs.org/breaking-changes/watch.html)

File: src/plugins/telemetryTable/components/table.vue [hunk 7]
```diff
@@ -630,16 +638,18 @@ export default {
       this.table.sortBy(this.sortOptions);
     },
     scroll() {
-      this.updateVisibleRows();
-      this.synchronizeScrollX();
+      this.$nextTick(() => {
```
Comment (scottbell) [inline src/plugins/telemetryTable/components/table.vue:641]: why do we have to wait for the next tick now?

File: src/ui/components/ObjectView.vue [hunk 8]
```diff
@@ -83,7 +83,8 @@ export default {
   },
   data() {
     return {
-      domainObject: this.defaultObject
+      domainObject: this.defaultObject,
+      viewKey: undefined
```
Comment (scottbell) [inline src/ui/components/ObjectView.vue:87]: seems semantically weird to assign undefined here, maybe `null` instead?

File: src/ui/layout/BrowseBar.vue [hunk 9]
```diff
@@ -179,7 +179,8 @@ export default {
       });
     },
     hasParent() {
-      return this.domainObject !== PLACEHOLDER_OBJECT && this.parentUrl !== '/browse';
+      return false;
+      // return this.domainObject !== PLACEHOLDER_OBJECT && this.parentUrl !== '/browse';
```
Comment (scottbell) [inline src/ui/layout/BrowseBar.vue:183]: What's the thinking behind this change?

File: src/ui/mixins/context-menu-gesture.js [hunk 10]
```diff
@@ -1,3 +1,6 @@
+import * as Emitter from 'tiny-emitter';
```
Comment (scottbell) [inline src/ui/mixins/context-menu-gesture.js:1]: why tiny-emitter vs the other Emitter used earlier?

File: src/api/composition/CompositionCollection.js (hunk 2 above)
Comment (ozyx) [reply src/api/composition/CompositionCollection.js:59]: To make a long story short, [Proxies and private members don't play well together](https://github.com/tc39/proposal-class-fields/issues/106).

File: src/ui/layout/BrowseBar.vue (hunk 9 above)
Comment (scottbell) [reply src/ui/layout/BrowseBar.vue:183]: @scottbell to look at this

**Comment by scottbell** [comment]
@scottbell also to look at why clicking on elements tab on OverlayPlots causes application to freeze.

File: src/ui/layout/BrowseBar.vue (hunk 9 above)
Comment (scottbell) [reply src/ui/layout/BrowseBar.vue:183]: It's because in `mct-tree.vue`, we call [calculateHeights](https://github.com/nasa/openmct/blob/vue3-compat-migration/src/ui/layout/mct-tree.vue#L304) which [loops forever](https://github.com/nasa/openmct/blob/vue3-compat-migration/src/ui/layout/mct-tree.vue#L942C14-L942C14) looking for a tree height. Changing `calculateHeights` to fire post the first Vue tick resolves this issue (and reveals more problems 😀)

Comment (scottbell) [reply src/ui/layout/BrowseBar.vue]: Fixed (and pushed) after also [toRaw'ing the proxied domainObject](https://github.com/nasa/openmct/blob/vue3-compat-migration/src/ui/layout/BrowseBar.vue#L183) in BrowseBar so the comparison would work.

**Comment by scottbell** [comment]
> @scottbell also to look at why clicking on elements tab on OverlayPlots causes application to freeze.

The issue here is that on mount in `PlotElementsPool`, we call [showSelection()](https://github.com/nasa/openmct/blob/master/src/plugins/inspectorViews/elements/PlotElementsPool.vue#L112).

This in turn [calls loadComposition()](https://github.com/nasa/openmct/blob/master/src/plugins/inspectorViews/elements/PlotElementsPool.vue#L146).

Which `InspectorViews` is listening to, who in turn [calls show()](https://github.com/nasa/openmct/blob/master/src/ui/inspector/InspectorViews.vue#L54) on the Inspector tabs.

Which in turn calls `mounted()` on the `PlotElementsViewProvider`, which calls `showSelection` and so on, and so on:
image

Haven't quite figured out how to break this loop, but obviously this doesn't happen with Vue2.

File: src/plugins/notebook/components/Notebook.vue [hunk 11]
```diff
@@ -975,7 +963,10 @@ export default {
       });
     },
     sectionsChanged({ sections, id = undefined }) {
+      this.sections = [...sections];
```
Comment (davetsay) [inline src/plugins/notebook/components/Notebook.vue:966]: Noice

File: src/plugins/folderView/components/GridItem.vue [hunk 12]
```diff
@@ -69,7 +70,7 @@ export default {
     }
   },
   methods: {
-    navigate() {
+    navigate(event) {
```
Comment (khalidadil) [inline src/plugins/folderView/components/GridItem.vue:73]: What is this event being used for?

Comment (ozyx) [reply src/plugins/folderView/components/GridItem.vue:73]: Looks like it's not. I'll change it to `_event` to notify it's available but unused

Comment (ozyx) [reply src/plugins/folderView/components/GridItem.vue:73]: Looks like it's not. I'll change it to `_event` to notify it's available but unused

File: src/plugins/plan/pluginSpec.js [hunk 13]
```diff
@@ -281,9 +281,9 @@ describe('the plugin', function () {
       });
     });
 
-    afterEach(() => {
-      component.$destroy();
-    });
+    // afterEach(() => {
```
Comment (akhenry) [inline src/plugins/plan/pluginSpec.js:284]: Commented code, we need this?

**Review by akhenry — CHANGES_REQUESTED** [review]

File: src/plugins/plot/MctPlot.vue [hunk 14]
```diff
@@ -395,13 +395,14 @@ export default {
     this.openmct.objectViews.on('clearData', this.clearData);
     this.$on('loadingComplete', this.loadAnnotations);
     this.openmct.selection.on('change', this.updateSelection);
-    this.setTimeContext();
-
     this.yAxisListWithRange = [this.config.yAxis, ...this.config.additionalYAxes];
 
-    this.loaded = true;
+    this.$nextTick(() => {
```
Comment (akhenry) [inline src/plugins/plot/MctPlot.vue:400]: Interesting, why was this needed to make Vue 3 work?

File: src/plugins/imagery/components/Compass/pluginSpec.js [hunk 15]
```diff
@@ -69,7 +69,7 @@ describe('The Compass component', () => {
   });
 
   afterAll(() => {
-    app.$destroy();
```
Comment (khalidadil) [inline src/plugins/imagery/components/Compass/pluginSpec.js:72]: Do we still need this?

File: src/plugins/telemetryTable/components/table.vue [hunk 16]
```diff
@@ -494,23 +501,24 @@ export default {
...
     this.openmct.time.on('bounds', this.boundsChanged);
 
     //Default sort
     this.sortOptions = this.table.tableRows.sortBy();
-    this.scrollable = this.$el.querySelector('.js-telemetry-table__body-w');
-    this.contentTable = this.$el.querySelector('.js-telemetry-table__content');
-    this.sizingTable = this.$el.querySelector('.js-telemetry-table__sizing');
-    this.headersHolderEl = this.$el.querySelector('.js-table__headers-w');
-    this.table.configuration.on('change', this.updateConfiguration);
+    this.scrollable = this.$refs.scrollable;
```
Comment (akhenry) [inline src/plugins/telemetryTable/components/table.vue:510]: 👍

File: src/plugins/telemetryTable/components/table.vue [hunk 17]
```diff
@@ -494,23 +501,24 @@ export default {
     this.table.tableRows.on('sort', this.updateVisibleRows);
     this.table.tableRows.on('filter', this.updateVisibleRows);
 
+    this.table.configuration.on('change', this.updateConfiguration);
```
Comment (akhenry) [inline src/plugins/telemetryTable/components/table.vue:504]: Why did this move?

**Review by akhenry — CHANGES_REQUESTED** [review]

File: src/plugins/telemetryTable/components/table.vue (hunk 17 above)
Comment (ozyx) [reply src/plugins/telemetryTable/components/table.vue:504]: I think I may have just moved it to be with the rest of the listeners. I can move it back if you think we should.

File: src/plugins/flexibleLayout/components/container.vue [hunk 18]
```diff
@@ -29,7 +29,7 @@
     <div
       v-show="isEditing"
       class="c-fl-container__header"
-      draggable="true"
+      :draggable="true"
```
[the remark this answers is not in the document]
Comment (akhenry) [reply src/plugins/flexibleLayout/components/container.vue:32]: Good catch. Yeah, this is just a regular HTML property, so the original usage is appropriate.

File: src/plugins/imagery/components/ImageryView.vue [hunk 19]
```diff
@@ -774,15 +754,41 @@ export default {
...
-      this.timeContext.on('timeSystem', this.trackDuration);
-      this.timeContext.on('clock', this.trackDuration);
+      this.timeContext.on('timeSystem', this.timeContextChanged);
+      this.timeContext.on('clock', this.timeContextChanged);
+      this.timeContextChanged();
     },
     stopFollowingTimeContext() {
       if (this.timeContext) {
-        this.timeContext.off('timeSystem', this.trackDuration);
-        this.timeContext.off('clock', this.trackDuration);
```
Comment (khalidadil) [inline src/plugins/imagery/components/ImageryView.vue:783]: Was there a trackDuration function that can be removed?

Comment (khalidadil) [reply src/plugins/imagery/components/ImageryView.vue:783]: oh I guess we're still using it elsewhere. Why are we switching from trackDuration here?

File: src/plugins/timeConductor/ConductorInputsFixed.vue [hunk 20]
```diff
@@ -30,7 +30,7 @@
         class="c-input--datetime"
         type="text"
         autocorrect="off"
-        spellcheck="false"
+        :spellcheck="false"
```
Comment (akhenry) [inline src/plugins/timeConductor/ConductorInputsFixed.vue:33]: This does not need to be dynamic, it will never change.

File: src/plugins/timeConductor/ConductorInputsFixed.vue [hunk 21]
```diff
@@ -54,7 +54,7 @@
...
```
Comment (akhenry) [inline src/plugins/timeConductor/ConductorInputsFixed.vue:57]: Does not need to be dynamic

File: src/plugins/telemetryTable/components/table.vue [hunk 22]
```diff
@@ -703,11 +713,13 @@ export default {
      * Calculates height based on total number of rows, and sets table height.
      */
     setHeight() {
-      let tableRowsLength = this.table.tableRows.getRowsLength();
-      this.totalHeight = this.rowHeight * tableRowsLength - 1;
-      // Set element height directly to avoid having to wait for Vue to update DOM
-      // which causes subsequent scroll to use an out of date height.
-      this.contentTable.style.height = this.totalHeight + 'px';
+      this.$nextTick(() => {
```
[the remark this answers is not in the document]
Comment (ozyx) [reply src/plugins/telemetryTable/components/table.vue:716]: It's not needed-- good catch!

**Review by davetsay — CHANGES_REQUESTED** [review]
Reviewed the following. A couple minor changes and questions. Great Work!

src/plugins/LADTable
src/plugins/latestDataClock
src/plugins/licenses
src/plugins/linkAction
src/plugins/localStorage
src/plugins/localTimeSystem
src/plugins/move
src/plugins/myItems
src/plugins/newFolderAction
src/plugins/notebook
src/plugins/notificationIndicator
src/plugins/objectMigration
src/plugins/openInNewTabAction

File: src/plugins/imagery/components/ImageryView.vue (hunk 19 above)
Comment (ozyx) [reply src/plugins/imagery/components/ImageryView.vue:783]: woah... good question?? let me look

Comment (ozyx) [reply src/plugins/imagery/components/ImageryView.vue:783]: Ah, it's part of this fix:
https://github.com/nasa/openmct/commit/344993ae2fd96fbfbeac5df597b0a817840699c9

File: src/plugins/viewDatumAction/ViewDatumAction.js [hunk 23]
```diff
@@ -21,7 +21,7 @@
  *****************************************************************************/
 
 import MetadataListView from './components/MetadataList.vue';
-import Vue from 'vue';
+import mount from 'utils/mount';
 
 export default class ViewDatumAction {
   constructor(openmct) {
```
Comment (khalidadil) [inline src/plugins/viewDatumAction/ViewDatumAction.js:27]: Can we chance _openmct to just openmct for consistency if we don't have a reason for the underscore?

File: src/plugins/notebook/components/Notebook.vue [hunk 24]
```diff
@@ -137,7 +137,10 @@
             @entry-selection="entrySelection(entry)"
           />
         </div>
-        <div v-if="showLockButton" class="c-notebook__commit-entries-control">
+        <div
+          v-if="isRestricted && filteredAndSortedEntries?.length > 0 && !selectedPage.isLocked"
```
[the remark this answers is not in the document]
Comment (ozyx) [reply src/plugins/notebook/components/Notebook.vue:141]: I whole-heartedly agree, but was running out of time to get that computed property reactive again

File: src/plugins/notebook/components/NotebookSnapshotContainer.vue [hunk 25]
```diff
@@ -43,7 +43,7 @@
       <span
         v-for="snapshot in snapshots"
         :key="snapshot.embedObject.id"
-        draggable="true"
+        :draggable="true"
```
[the remark this answers is not in the document]
Comment (ozyx) [reply src/plugins/notebook/components/NotebookSnapshotContainer.vue:46]: nay

File: src/plugins/notificationIndicator/components/NotificationIndicator.vue [hunk 26]
```diff
@@ -79,7 +79,7 @@ export default {
       this.showNotificationsOverlay = flag;
     },
     updateNotifications() {
-      this.notifications = this.openmct.notifications.notifications;
+      this.notifications = [...this.openmct.notifications.notifications];
```
[the remark this answers is not in the document]
Comment (ozyx) [reply src/plugins/notificationIndicator/components/NotificationIndicator.vue:82]: To be honest, I haven't figured out how to make the notifications API reactive yet. This is creating a new array and seemed to allow `this.notifications` to pick up the change and be reactive. But the Notifications Indicator is still pretty buggy due to reactivity issues.

File: src/plugins/viewDatumAction/ViewDatumAction.js (hunk 23 above)
Comment (ozyx) [reply src/plugins/viewDatumAction/ViewDatumAction.js:27]: ya

File: src/plugins/condition/pluginSpec.js [hunk 27]
```diff
@@ -256,7 +256,7 @@ describe('the plugin', function () {
     });
 
     afterEach(() => {
-      component.$destroy();
```
Comment (khalidadil) [inline src/plugins/condition/pluginSpec.js]: Leaving a note so we can track this to fix later

File: src/ui/layout/AppLogo.vue [hunk 28]
```diff
@@ -25,29 +25,37 @@
...
+          template: '<about-dialog></about-dialog>'
         },
-        template: '<about-dialog></about-dialog>'
-      }).$mount();
+        {
+          app: this.openmct.app
+        }
+      );
+
+      el.classList.add('u-contents');
```
Comment (akhenry) [inline src/ui/layout/AppLogo.vue:53]: y tho?

File: src/ui/layout/RecentObjectsListItem.vue [hunk 29]
```diff
@@ -35,7 +35,7 @@
         ref="recentObjectName"
         class="c-recentobjects-listitem__title"
         :name="domainObject.name"
-        draggable="true"
+        :draggable="true"
```
Comment (akhenry) [inline src/ui/layout/RecentObjectsListItem.vue]: Does not need to be dynamic

File: src/ui/components/ObjectLabel.vue [hunk 30]
```diff
@@ -21,6 +21,7 @@
 -->
 <template>
   <a
+    ref="root"
```
Comment (khalidadil) [inline src/ui/components/ObjectLabel.vue:24]: Why do we need this again?

File: src/plugins/plot/MctPlot.vue (hunk 14 above)
Comment (ozyx) [reply src/plugins/plot/MctPlot.vue]: Eventually this ends up interacting with a $ref, so need `nextTick()` here to ensure it's available

File: src/ui/components/ObjectView.vue [hunk 31]
```diff
@@ -272,9 +272,7 @@ export default {
         this.loadComposition();
       }
 
-      this.viewContainer = document.createElement('div');
-      this.viewContainer.classList.add('l-angular-ov-wrapper');
```
Comment (khalidadil) [inline src/ui/components/ObjectView.vue:276]: Does this class not have any associated styles? I know at some point we had some old classes with styles that were still needed.

Comment (ozyx) [reply src/ui/components/ObjectView.vue:276]: I removed this with @charlesh88. It’s old and no longer needed

File: src/ui/mixins/context-menu-gesture.js [hunk 32]
```diff
@@ -15,23 +17,29 @@ export default {
   },
   mounted() {
     //TODO: touch support
-    this.$el.addEventListener('contextmenu', this.showContextMenu);
+    this.$nextTick(() => {
```
Comment (khalidadil) [inline src/ui/mixins/context-menu-gesture.js:20]: Why do we need this?

File: src/ui/components/ObjectLabel.vue (hunk 30 above)
Comment (ozyx) [reply src/ui/components/ObjectLabel.vue:24]: This works in conjunction with context-menu-gesture mixin

File: src/ui/preview/Preview.vue [hunk 33]
```diff
@@ -156,9 +156,7 @@ export default {
       delete this.existingViewElement;
     },
     initializeViewContainer() {
-      this.viewContainer = document.createElement('div');
-      this.viewContainer.classList.add('l-angular-ov-wrapper');
```
Comment (khalidadil) [inline src/ui/preview/Preview.vue:160]: Same question about the class here

Comment (ozyx) [reply src/ui/preview/Preview.vue:160]: No longer needed according to @charlesh88

File: src/ui/toolbar/Toolbar.vue [hunk 34]
```diff
@@ -138,7 +143,7 @@ export default {
         domainObject,
         '*',
         function (newObject) {
-          this.domainObjectsById[id].newObject = JSON.parse(JSON.stringify(newObject));
```
Comment (khalidadil) [inline src/ui/toolbar/Toolbar.vue:141]: I wonder why we were doing this?

File: src/ui/mixins/context-menu-gesture.js (hunk 32 above)
Comment (ozyx) [reply src/ui/mixins/context-menu-gesture.js:20]: Vue 3 changes the way refs work. We need the nextTick to ensure it's available.

Comment (ozyx) [reply src/ui/mixins/context-menu-gesture.js:20]: Vue 3 changes the way refs work. We need the nextTick to ensure it's available.

File: src/ui/toolbar/Toolbar.vue (hunk 34 above)
Comment (ozyx) [reply src/ui/toolbar/Toolbar.vue:141]: 🤷‍♂️

we shouldn't have been

Comment (ozyx) [reply src/ui/toolbar/Toolbar.vue:141]: 🤷‍♂️

we shouldn't have been

File: src/ui/layout/search/GrandSearchSpec.js [hunk 35]
```diff
@@ -204,7 +204,7 @@ xdescribe('GrandSearch', () => {
   afterEach(() => {
     openmct.objects.inMemorySearchProvider.worker = sharedWorkerToRestore;
     openmct.router.path = originalRouterPath;
-    grandSearchComponent.$destroy();
+    //grandSearchComponent.$destroy();
```
Comment (akhenry) [inline src/ui/layout/search/GrandSearchSpec.js]: This looks like it should not be commented out.

File: src/ui/layout/search/ObjectSearchResult.vue [hunk 36]
```diff
@@ -36,7 +36,7 @@
         ref="resultName"
         class="c-gsearch-result__title"
         :name="resultName"
-        draggable="true"
+        :draggable="true"
```
Comment (akhenry) [inline src/ui/layout/search/ObjectSearchResult.vue]: Does this need to be dynamic?

File: src/plugins/displayLayout/components/DisplayLayout.vue [hunk 37]
```diff
@@ -223,9 +221,13 @@ export default {
     this.composition.load();
     this.gridDimensions = [this.$el.offsetWidth, this.$el.scrollHeight];
 
+    this.openmct.objects.observe(this.domainObject, 'configuration.items', (items) => {
```
Comment (davetsay) [inline src/plugins/displayLayout/components/DisplayLayout.vue:224]: okay okay

**Review by davetsay — APPROVED** [review]

File: src/ui/toolbar/Toolbar.vue (hunk 34 above)
Comment (khalidadil) [reply src/ui/toolbar/Toolbar.vue:141]: yeah, I've ripped out hacky stuff before and found out later why it needed to be there 🥲

If there was some reason, this will force us to find it!

File: src/plugins/viewDatumAction/ViewDatumAction.js (hunk 23 above)
Comment (khalidadil) [reply src/plugins/viewDatumAction/ViewDatumAction.js:27]: thanks!

**Review by khalidadil — APPROVED** [review]
Reviewed the files I indicated on the spreadsheet - good work on this huge change!!
